apps-rendering/src/campaign.ts

```diff
@@ -11,23 +11,17 @@ const getCallout = (
 	id: string,
 	campaigns: Campaign[],
 ): Optional<ParticipationFields> => {
-	const [campaign, ...tail] = campaigns;
-
-	// If the array is empty, `campaign` will be `undefined`. The linter doesn't
-	// know this without `noUncheckedIndexedAccess` turned on.
-	// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- See above.
-	if (campaign === undefined) {
+	if (campaigns.length === 0) {
 		return Optional.none();
 	}
 
-	if (
-		campaign.fields.kind === 'callout' &&
-		campaign.fields.callout.tagName === id
-	) {
+	const campaign = campaigns.find((campaign) => campaign.id === id);
+
+	if (campaign?.fields.kind === 'callout') {
 		return Optional.some(campaign.fields.callout);
 	}
 
-	return getCallout(id, tail);
```

Comment:
As we are using the campaign id now, we can search the campaigns for the specific campaign and no longer need this to be recursive